enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+
+The PO manager will manage a ClusterOperator resource, which will act as a proxy for communicating the aggregate health of the PO packages back to the CVO. Future phases may require PO packages to reconcile OLM's OperatorCondition resource, or directly manage a ClusterOperator resource so the PO manager can properly relay package state back to the CVO.
+
+### Implementation Details
+
+> Note: The following sub-headers may frequently refer to OLM 1.x terminology. It's recommended to read through the ["Common Terminology"](#Common-Terminology) header for those less familiar with recent OLM developments.
+
+#### PO Manager
+
+The PO manager would abstract the process of sourcing registry+v1 bundles from catalog(s), chosing the "best bundle" to install for each desired package, and ensuring those packages get installed on the cluster.
+
+In order to determine the set of bundles that need to be installed, the PO manager can interact with the default, `redhat-operators` CatalogSource and use OLM's existing registry gRPC APIs to list the available bundles from that catalog.
+
+> The OLM marketplace component is now an optional CVO component. Admins can disable this component using the component capabilities functionality introduced in previous 4.x releases. Available workarounds include admins creating their own CatalogSource resources. This behavior may require explicit documentation, or expanding the phase 0 design scope to be able to accommodate all the catalog sources in the cluster.
+
+After determining the set of bundles that need to exist in order to satify admin package intent, the PO manager's application layer will need to attempt to install those bundles. The failed deployment of a bundle should be reflected in the aggregate ClusterOperator resource.
+
+Admins may elect to install or delete a PO package after cluster creation. This process would include creating, or deleting, a PlatformOperators custom resource. For deletion, the PO manager's application layer would be responsible for placing owner references on the underlying bundle deployment resources.
+
+**Note**: When an underlying bundle deployment is deleted, that deletion cascades to all of the bundle manifests. This may include deleting cluster-scoped resources, e.g. CRDs, which can impact workload health in unsupported configurations. This behavior differs from the current OLM 0.x behavior, due to an opinionated tenancy model it needs to protect.
+
+After a bundle has been deployed to the cluster by the PO manager, admins may need to perform manual steps for upgrading that bundle version.
+
+That process may require the following steps:
+
+- Find the current bundle version for the configured PO package name.
+- Verifying that the redhat-operators index image contains a newer, higher semver bundle than what's currently applied to the cluster.
+- Deleting the underlying bundle deployment resources to force the PO manager to select a new bundle.
```

Comment:
This may be a candidate for some re-organization, but an admin may take the following steps to remove a PO from the cluster:

- List the PlatformOperators custom resources in the cluster
- Delete the PlatformOperators custom resources that matches the package name they wish to delete
- And, the PO manager will take care of the rest.

Given we place owner references on the underlying BundleDeployment where the PlatformOperators custom resource is the owner, deleting that custom resource will result in the BundleDeployment being deleted. What's worth mentioning here is that the BundleDeployment _also_ places owner references on the resources it manages, which implies a cascading deletion of bundle contents. I _think_ outlines this behavior further down in the proposal, so maybe this is already covered, but it's not mentioned earlier enough?

Reply:
Updated this language in the latest commit, and add a note around cascading deletion of bundle contents.